src/slurm_plugin/clustermgtd.py

```diff
@@ -964,6 +973,73 @@ def _is_node_in_replacement_valid(self, node, check_node_is_valid):
             return not time_is_expired if check_node_is_valid else time_is_expired
         return False
 
+    def _handle_ice_nodes(self, active_nodes, unhealthy_dynamic_nodes):
+        ice_nodes = [node for node in unhealthy_dynamic_nodes if node.is_ice()]
+        self.update_insufficient_capacity_compute_resources()
+        self.reset_timeout_expire_compute_resources(ice_nodes)
+        self._set_ice_crs_to_down(active_nodes)
+
+    def update_insufficient_capacity_compute_resources(self, ice_nodes):
+        """Add compute resource to insufficient_capacity_compute_resources if node is ICE node."""
+        for node in ice_nodes:
+            compute_resource = node.get_compute_resource_name()
+            if compute_resource not in self._insufficient_capacity_compute_resources:
+                self._insufficient_capacity_compute_resources[compute_resource] = {
+                    "ice_time": self._current_time,
+                    "error_code": node.error_code,
+                }
```

Comment:
would it make sense to use a class for this? We could also consider keeping support only for Python>=3.7 in node and use data classes https://realpython.com/python-data-classes/.

What I mean is to encode the structure of:
```
{
                    "ice_time": self._current_time,
                    "error_code": node.error_code,
                }
```
inside a data class and eventually generalize the concept. For example the class can be named `InstanceFailureData` or `InstanceFailureEvent` and contain 2 attributes: error_code and timestamp

Reply:
Yes, added ComputeResourceFailureEvent